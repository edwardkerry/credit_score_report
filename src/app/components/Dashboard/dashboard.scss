body {
  background-image: url('/background--dashboard.png');
  background-repeat: no-repeat;
  background-position: center;
  background-attachment: fixed;
  background-size: cover;
  height:100%;
  width:100%;
}