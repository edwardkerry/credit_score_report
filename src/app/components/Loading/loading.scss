.report__loading {
  width: 100px;
  height: 100px;
  background-size: cover;
  -webkit-animation: fadein 2s ease-in alternate infinite;
  -moz-animation: fadein 2s ease-in alternate infinite;
  animation: fadein 2s ease-in alternate infinite;
  background-image: url('/clearscore.png');
}

@-webkit-keyframes fadein {
  from { opacity: 0.2; }
  to { opacity: 0.8; }
}

@-moz-keyframes fadein {
  from { opacity: 0.2; }
  to { opacity: 0.8; }
}

@keyframes fadein {
  from { opacity: 0.2; }
  to { opacity: 0.8; }
}