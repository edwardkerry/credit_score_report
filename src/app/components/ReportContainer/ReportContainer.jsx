import React, { Component } from 'react';
import {shape, number, string } from 'prop-types';

import Carousel from 'nuka-carousel';
import CreditScoreReport from '../CreditScoreReport/CreditScoreReport.jsx';
import LongTermDebtReport from '../LongTermDebtReport/LongTermDebtReport.jsx';

import bemHelper from '../../utils/bem';

const cn = bemHelper({ block: 'report' });

const ReactContainer = ({ creditScore, longTermDebt}) => {

  const { score, maxScore, description } = { ...creditScore };
  const { currentCreditLimit, currentDebt, debtChange} = { ...longTermDebt }

   return (
    <Carousel className={cn('carousel', null)}>
      <CreditScoreReport 
        score={score} 
        maxScore={maxScore} 
        description={description} 
      />
      <LongTermDebtReport 
        currentCreditLimit={currentCreditLimit} 
        currentDebt={currentDebt} 
        debtChange={debtChange}
      />
    </Carousel>
  );
};

ReactContainer.propTypes = {
  creditScore: shape({
    description: string.isRequired,
    maxScore: number.isRequired,
    score: number.isRequired 
  }),

  longTermDebt:  shape({
    currentDebt: number.isRequired,
    currentCreditLimit: number.isRequired,
    debtChange: number.isRequired
  })
}

export default ReactContainer;
