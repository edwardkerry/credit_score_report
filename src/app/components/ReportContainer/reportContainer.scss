.report__container{
  display: flex;
  margin-top: 20vh;
  margin-bottom: 25vh;
  border: solid rgba(255, 255, 255, 0.932) 2px;
  border-radius: 50%;
  background: rgba(110, 107, 107, 0.5);
  height: 400px;
  width: 400px;
  overflow: hidden;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 25px;
  color: white;
  text-align: center;
  
  @media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
    font-size: 15px;
    height: 250px;
    width: 250px;
  }

  .slider-list {
    margin-top: 75px !important;
    @media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
      margin-top: 50px !important;
    }
  }
}