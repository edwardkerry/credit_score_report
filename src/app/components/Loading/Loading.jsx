import React from 'react';

import bemHelper from '../../utils/bem';

import './loading.scss';

const cn = bemHelper({ block: 'content' });

export default() => (
  <div className={cn('loading', null)} />
)