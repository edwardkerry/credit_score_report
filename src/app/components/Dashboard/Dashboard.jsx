import React, { Component } from 'react'
import { getJSON } from '../../utils'
import config from  '../../../config/environment.js'
import makeCreditScore from '../../models/creditScore'
import makeLongTermDebt from '../../models/longTermDebt'
import Loading from '../Loading/Loading'
import ReportContainer from '../ReportContainer/ReportContainer'
import './dashboard.scss';

class Dashboard extends Component {

  constructor(props) {
    super(props)
    this.state = {
      loading: true,
      creditScore: {},
      longTermDebt: {}
    }
  }

  componentDidMount() {
    getJSON(`${config.api.host}/creditReportInfo.json`)
    .then((response) => {
      this.setState({
        loading: false,
        creditScore: makeCreditScore(response.creditReportInfo),
        longTermDebt: makeLongTermDebt(response.creditReportInfo)
      })
    })
  }

   render() {
    return this.state.loading ?
      <Loading /> :
      <ReportContainer creditScore={this.state.creditScore} longTermDebt={this.state.longTermDebt}/>
  }
}

export default Dashboard;
