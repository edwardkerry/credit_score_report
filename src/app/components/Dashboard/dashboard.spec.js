import React from 'react';
import renderer from 'react-test-renderer';

import { creditScoreReport, longTermDebtReport } from '../../../../tests/mocks/fixtures'
import mockAPIResponse from '../../../../tests/mocks/mockAPIResponse'

import Dashboard from './Dashboard.jsx';
import Loading from '../Loading/Loading.jsx'
import ReportContainer from '../ReportContainer/ReportContainer.jsx'

describe('Dashboard', () => {
  describe('when loading', () => {
    it('displays a loading sign', () => {
      const component = renderer.create(<Dashboard />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe('when loaded', () => {
    it('displays a report container', () => {
      const component = renderer.create(<Dashboard loading={false} />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });

    
    it('passes state to children as props', () => {
      const component = mount(<Dashboard />);
  
      component.setState({
        loading: false,
        creditScore: creditScoreReport,
        longTermDebt: longTermDebtReport,
      });
  
      const result = component.find(ReportContainer);
      expect(result.prop('creditScore')).toEqual(creditScoreReport);
      expect(result.prop('longTermDebt')).toEqual(longTermDebtReport);
    });
  });

  xdescribe('componentDidMount', () => {

    it('sets loading to false after fetching data', (done) => {
      const component = shallow(<Dashboard />);
      component.instance().componentDidMount();
      expect(component.state('loading')).toEqual(false);
    });
  });


});