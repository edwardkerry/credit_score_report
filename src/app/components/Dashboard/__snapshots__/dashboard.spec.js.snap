// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard when loaded displays a report container 1`] = `
<div
  className="report__loading"
/>
`;

exports[`Dashboard when loading displays a loading sign 1`] = `
<div
  className="report__loading"
/>
`;
